superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -1154,7 +1154,7 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
   }, [passwordsNeeded]);
 
   useEffect(() => {
-    if (db) {
+    if (db && isSSHTunneling) {
       setUseSSHTunneling(!isEmpty(db?.ssh_tunnel));
     }
   }, [db]);
```

Comment:
```suggestion
    if (db && isSSHTunneling) {
      setUseSSHTunneling(!isEmpty(db?.ssh_tunnel));
    }
  }, [db, isSSHTunneling]);
```
Adding `isSSHTunneling` to the dependency array will remove the dependency warning.